File: Tutorials/PHP/02_Serial_IntegerDecoder/serial.php
<?php
  // Read from serial with data coming from RFM12PI with RFM12_Demo sketch 
  // All Emoncms code is released under the GNU Affero General Public License.

  $c = stream_context_create(array('dio' =>
    array('data_rate' => 9600,
          'data_bits' => 8,
          'stop_bits' => 1,
          'parity' => 0,
          'flow_control' => 0,
          'is_canonical' => 1)));

  if (PATH_SEPARATOR != ";") {
    $filename = "dio.serial:///dev/ttyUSB0";
  } else {
    $filename = "dio.serial://dev/ttyUSB0";
  }

  $f = fopen($filename, "r+", false, $c);
  stream_set_timeout($f, 0,1000);

  while(true)
  {
    $data = fgets($f);
    if ($data && $data!="\n")
    {
      echo "Raw data: $data";
      $parts = explode(' ',$data);
      $nodeid = (int) $parts[1];

      $values = array();
      for($i=2; $i<(count($parts)-1); $i+=2)
      {
        // Get 16-bit integer
        $int16 = $parts[$i] + $parts[$i+1]*256;
        if ($int16>32768) $int16 = -65536 + $int16;
        $value = $int16;
        $values[] = $value;
      }
      echo "  Decoded Node: $nodeid:[".implode($values,',')."]\n";
    }
  }

  fclose($f);

File: Tutorials/Python/02_Serial_IntegerDecoder/serial.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

# Read from serial with data coming from RFM12PI with RFM12_Demo sketch 
# All Emoncms code is released under the GNU Affero General Public License.

import serial, sys, string
import time, datetime

# Set this to the serial port of your emontx and baud rate, 9600 is standard emontx baud rate
ser = serial.Serial('/dev/ttyAMA0', 9600)

while 1:

  # Read in line of readings from emontx serial
  linestr = ser.readline()

  # Remove the new line at the end
  linestr = linestr.rstrip()

  print "DATA RX:"+linestr

  # Split the line at the whitespaces
  values = linestr.split(' ')
  nodeid = int(values[1])
  nameid = 1

  for i in range(2,(len(values)-1),2):

    # Get 16-bit integer
    value = int(values[i]) + int(values[i+1])*256
    if value>32768: value -= -65536

    print str(nodeid)+" "+str(nameid)+" "+str(value)
   
    nameid += 1
